internal/execution/execution.go

```diff
@@ -114,9 +116,26 @@ func call(ctx context.Context, url string, timeout time.Duration, body []byte) (
 	}
 	defer resp.Body.Close()
 
+	return handleResponse(resp)
+}
+
+func handleResponse(resp *http.Response) ([]byte, error) {
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
 	// Check for success between 200 and 299, redirect 300 to 399 is handled by the client, return error with statusCode >= 400
 	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
-		return io.ReadAll(resp.Body)
+		return data, nil
 	}
-	return nil, zerrors.ThrowUnknown(nil, "EXEC-dra6yamk98", "Errors.Execution.Failed")
+
+	return nil, zhttp.HTTPStatusCodeToZitadelError(
+		zerrors.ThrowError(nil,
+			"EXEC-1n27xlas",
+			fmt.Sprintf("status: %d, body: '%s'", resp.StatusCode, string(data)),
```

Comment:
Why do we introduce a custom format which no parser will be able to parse? I highly recommend an object which can be marshalled to json or proto analog the example of the previous reviews comment

Reply:
I changed it to a struct which is the marshalled into the error, before it was in an error chain, and before that it was also in error as a parent of the error. I struggle to understand what the difference here should be as the information is available as string objet in all occasions, besides now its a json as string in an object.

Reply:
The string looks more or less correct now. IMO it should be `statusCode` and not `status_code`. additionally It should be an object and not an object marshaled into a string

Reply:
`statusCode` I get, but you sure about the object? I mean if the object is not marshalled you get the representation like `Parent=({403 {"my": "value"}})` of the object, and not the json with the attributenames `Parent=({"statusCode": 403, "body": {"my": "value"}})`

Reply:
By the way, why should it be a parent-error? They don't get marshalled neither in the logs, nor in if you get an error response.